Lesson 4

Hermann27 opened the pull request:
Please, review my lesson 4 assignment and let me know if you have any questions or suggestions.
Big thank you for giving me the advance to well structure my project on the last pull request.

Hermann27 committed:
clean up some file

Hermann27 committed:
Installed Prop Types Library

Hermann27 committed:
On This commit, I have installed Proptypes using this cmd : npm install prop-types.

The had to implement the Protypes on all my components

Hermann27 committed:
Change my Proptype emplacement.

AnnaSolovykh commented on src/index.css:
```diff
@@ -1,9 +1,65 @@
+  padding: 0;
+  font-family: "Poppins", sans-serif;
+}
+/*------------------sidemenu--------------------*/
```
I noticed that there is inconsistent naming in the CSS variables. For example, here you're using snake_case format. Since camelCase is used elsewhere in the app, we should apply that consistently across all styles for better readability and maintainability.

Additionally:

1. Since we're using CSS modules, it's important to place each module CSS file in the same folder as the corresponding JSX file. This will keep the component-specific styles organized.

2. The index.css file should only contain global styles that apply across the entire app, such as resets, typography, or general layout settings. Any component-specific styles should go into their respective CSS module files.

Hermann27 replied:
Thanks for the feedback.

Hermann27 committed:
Update CSS style on index.css amd TodoListItem.module.css file

AnnaSolovykh commented on index.html:
```diff
@@ -2,12 +2,37 @@
   <body>
     <div id="root"></div>
     <script type="module" src="/src/main.jsx"></script>
+    <script
```
Bootstrap's JavaScript files should not be loaded via CDN in a Vite app because Vite has its own bundling process. Instead of using CDN links for Bootstrap’s JS bundle, you should import the Bootstrap JS files via npm as shown above.

AnnaSolovykh commented on index.html:
```diff
@@ -2,12 +2,37 @@
+      referrerpolicy="no-referrer"
+    />
+
+    <!-- App icon -->
```
This will work in development, but in production, this might break depending on the structure of your build output. It's better to store such assets in a public folder that Vite recognizes.

Hermann27 committed:
Cleanup my index.html to remove anything that was added.

AnnaSolovykh commented on index.html:
```diff
@@ -2,12 +2,37 @@
+      crossorigin="anonymous"
+    />
+
+    <!-- font awsome cdn -->
```
While using Bootstrap and Font Awesome via a CDN is fine, it's not ideal for a Vite project where module bundling is a key feature. Instead, you should import these libraries directly into your project using npm, which makes version management easier and enables tree-shaking to avoid loading unnecessary CSS/JS.

Using npm helps lock versions and prevents unintentional updates.
Vite can bundle only the necessary parts of these libraries, reducing load times.

AnnaSolovykh commented:
Please revert index.html to the initial view (check it on main) and delete unrelated css in index.css file

Hermann27 committed:
On this commit, I have implemented:
1. Sort my data from my airtable using JavaScript inside the fectData func.

2. Created a toggle button so the user can switch between asc and desc sort order.

3.Automatically sorted my data when added.

Hermann27 committed:
Following changed have been added on this commit:
1. use css module on the app.css to center my cotainer.

2.removed the redomdance of useEffect on App.jxs

3. Refactored the code (re-structure and removed unnecessary code that was commented).

Hermann27 committed:
This updates fixed the mixing createdTime field on the display Table.

Hermann27 committed:
This commit resolved:
1. Formatting the Create Time field for a better format.

2. Update some style button.
3. Added an update button only.
